homeassistant/components/nfandroidtv/notify.py

```diff
@@ -14,115 +13,69 @@
-    "top-right": 2,
-    "top-left": 3,
-    "center": 4,
-}
-
-TRANSPARENCIES = {
-    "default": 0,
-    f"0{PERCENTAGE}": 1,
-    f"25{PERCENTAGE}": 2,
-    f"50{PERCENTAGE}": 3,
-    f"75{PERCENTAGE}": 4,
-    f"100{PERCENTAGE}": 5,
-}
-
-COLORS = {
-    "grey": "#607d8b",
-    "black": "#000000",
-    "indigo": "#303F9F",
-    "green": "#4CAF50",
-    "red": "#F44336",
-    "cyan": "#00BCD4",
-    "teal": "#009688",
-    "amber": "#FFC107",
-    "pink": "#E91E63",
-}
+_LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_DURATION, default=DEFAULT_DURATION): vol.Coerce(int),
-        vol.Optional(CONF_FONTSIZE, default=DEFAULT_FONTSIZE): vol.In(FONTSIZES.keys()),
-        vol.Optional(CONF_POSITION, default=DEFAULT_POSITION): vol.In(POSITIONS.keys()),
-        vol.Optional(CONF_TRANSPARENCY, default=DEFAULT_TRANSPARENCY): vol.In(
-            TRANSPARENCIES.keys()
+# Deprecated in Home Assistant 2021.8
+PLATFORM_SCHEMA = cv.deprecated(
+    vol.All(
+        PLATFORM_SCHEMA.extend(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Optional(CONF_DURATION): vol.Coerce(int),
+                vol.Optional(CONF_FONTSIZE): vol.In(Notifications.FONTSIZES.keys()),
+                vol.Optional(CONF_POSITION): vol.In(Notifications.POSITIONS.keys()),
+                vol.Optional(CONF_TRANSPARENCY): vol.In(
+                    Notifications.TRANSPARENCIES.keys()
+                ),
+                vol.Optional(CONF_COLOR): vol.In(Notifications.BKG_COLORS.keys()),
+                vol.Optional(CONF_TIMEOUT): vol.Coerce(int),
+                vol.Optional(CONF_INTERRUPT): cv.boolean,
+            }
         ),
-        vol.Optional(CONF_COLOR, default=DEFAULT_COLOR): vol.In(COLORS.keys()),
-        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): vol.Coerce(int),
-        vol.Optional(CONF_INTERRUPT, default=DEFAULT_INTERRUPT): cv.boolean,
-    }
+    )
 )
 
 
-def get_service(hass, config, discovery_info=None):
-    """Get the Notifications for Android TV notification service."""
-    remoteip = config.get(CONF_HOST)
-    duration = config.get(CONF_DURATION)
-    fontsize = config.get(CONF_FONTSIZE)
-    position = config.get(CONF_POSITION)
-    transparency = config.get(CONF_TRANSPARENCY)
-    color = config.get(CONF_COLOR)
-    interrupt = config.get(CONF_INTERRUPT)
-    timeout = config.get(CONF_TIMEOUT)
-
+async def async_get_service(hass: HomeAssistant, config, discovery_info=None):
+    """Get the NFAndroidTV notification service."""
+    if discovery_info is not None:
+        notify = await hass.async_add_executor_job(
+            Notifications, discovery_info[CONF_HOST]
+        )
+        return NFAndroidTVNotificationService(
+            notify,
+            hass.config.is_allowed_path,
+        )
+    notify = await hass.async_add_executor_job(Notifications, config.get(CONF_HOST))
```

Comment:
Log a warning and import the config here to a config flow instead. Then return None.